### DISCUSSION TITLE
Fund function giving error

### DISCUSSION THREAD

**Discussion by Lucian-Ryu** [Cyfrin/foundry-full-course-cu#247] [Q&A] [answered]
hello, im testing the FundMe.sol contract on remix and when i try using the fund function is giving me an error where the transaction dosnt go trough. it says the function should be payable, but it is.

am i missing something here?

here is my code

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {PriceConverter} from "./PriceConverter.sol";

contract FundMe{

using PriceConverter for uint256;

uint256 public minimumUsd  = 5e18;

address[] public funders;
mapping(address funder => uint256 amountFunded) public addressToAmountFunded;

address public owner;

constructor() {
    owner = msg.sender;
}

modifier onlyOwner() {
    require(msg.sender == owner, "Sender is not owner");
    _;
}

function fund() public payable {
   require (msg.value.getConversionRate() >= minimumUsd, "didnt send enough ETH");  //1e18 = 1 ETH
   funders.push(msg.sender);
   addressToAmountFunded[msg.sender] +=  msg.value;
}
function withdraw() public onlyOwner {
for(uint256 funderIndex = 0; funderIndex < funders.length; funderIndex++){
address funder = funders[funderIndex];
addressToAmountFunded[funder] = 0;
}
funders = new address;
(bool callSuccess, ) = payable(msg.sender).call{value: address(this).balance}("");
require(callSuccess, "Call failed");

}

}

**Comment by shikhar229169** [comment]
Can you provide the detailed screenshot of the error you received on remix.

**Reply by Lucian-Ryu** [reply]
![2023-07-05](https://github.com/Cyfrin/foundry-full-course-f23/assets/114004261/ed3a52df-882f-44b0-ad97-7525452a848f)


thanks for your answer, here is the screenshot of the error

**Comment by asobiesk** [comment]
Most likely you are deploying to local EVM instead of Sepolia testnet. The issue is that PriceConverter tries to call **latestRoundData** method on the contract at a given address, but this contract does not exist on the local chain. This issue will be addressed later in the course when you'll be mocking Price Feed on your local Anvil chain, but in Remix if you want to get it working pick "Injected Provider - MetaMask".

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/26230946/c58c5497-d04a-4ed7-b27e-ed8eb62e55b6)

**Comment by pyromaus** [comment, marked as answer]
```solidity
require (msg.value.getConversionRate() >= minimumUsd, "didnt send enough ETH"); //1e18 = 1 ETH
```
comment this line out and it'll work. you need a mock contract or a valid price feed

**Reply by Lucian-Ryu** [reply]
thank you it worked.

**Reply by Loupsteb** [reply]
Thank you very much, i stuck on this for 2 hours

**Comment by Aishwaria8786** [comment]
You can add ``` before and after the code. It will make your code more readable to understand what went wrong.

**Reply by Lucian-Ryu** [reply]
thanks, will do that in the future.
